From stable-bounces@linux.kernel.org Fri Feb 16 15:00:49 2007
From: Stephen Hemminger <shemminger@linux-foundation.org>
Date: Fri, 16 Feb 2007 14:56:10 -0800
Subject: sky2: dont flush good pause frames
To: stable@kernel.org
Message-ID: <20070216225836.148603000@linux-foundation.org>
Content-Disposition: inline; filename=sky2-pause-flush.patch

From: Stephen Hemminger <shemminger@linux-foundation.org>

Don't mark pause frames as errors. This problem caused transmitter not
to pause and would effectively take out a gigabit switch because the
it can't handle overrun.

Signed-off-by: Stephen Hemminger <shemminger@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/sky2.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.20.1.orig/drivers/net/sky2.h
+++ linux-2.6.20.1/drivers/net/sky2.h
@@ -1579,7 +1579,7 @@ enum {
 
 	GMR_FS_ANY_ERR	= GMR_FS_RX_FF_OV | GMR_FS_CRC_ERR |
 			  GMR_FS_FRAGMENT | GMR_FS_LONG_ERR |
-		  	  GMR_FS_MII_ERR | GMR_FS_GOOD_FC | GMR_FS_BAD_FC |
+		  	  GMR_FS_MII_ERR | GMR_FS_BAD_FC |
 			  GMR_FS_UN_SIZE | GMR_FS_JABBER,
 };
 
